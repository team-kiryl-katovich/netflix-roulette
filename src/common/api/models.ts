export type Result<T> = {
  data?: T;
  error?: Error;
};

export type Offset = {
  offset?: number;
  limit?: number;
};

export type OffsetResult<T> = Result<T> &
  Offset & {
    total?: number;
  };

export type Movie = {
  id: number;
  title: string;
  tagline: string;
  vote_average: number;
  vote_count: number;
  release_date: string;
  poster_path: string;
  overview: string;
  budget: number;
  revenue: number;
  runtime: number;
  genres: string[];
};

export type SortBy = keyof Movie;
export type SortOrder = 'desc' | 'asc';
export type SearchBy = Extract<keyof Movie, 'title' | 'genres'>;

export type MovieFilter = {
  sortBy?: SortBy;
  sortOrder?: SortOrder;
  search?: string;
  searchBy?: SearchBy;
  filter?: string;
};
