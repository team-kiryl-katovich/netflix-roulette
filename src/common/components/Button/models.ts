export type ButtonProps = {
  title: string;
  className?: string;
  type?: 'primary' | 'secondary';
  onClick: () => void;
};
