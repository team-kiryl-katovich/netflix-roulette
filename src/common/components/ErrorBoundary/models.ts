export type ErrorBoundaryProps = {
  children: React.ReactNode;
};

export type ErrorBoundaryState = {
  error: Error;
  errorInfo: React.ErrorInfo;
};
