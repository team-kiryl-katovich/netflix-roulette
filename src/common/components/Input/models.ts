export type InputProps = {
  placeholder?: string;
  className?: string;
  onChange: (val: string) => void;
};
