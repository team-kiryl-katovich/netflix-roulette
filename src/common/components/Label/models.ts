export type LabelProps = {
  text: string;
  className?: string;
  uppercase?: boolean;
};
