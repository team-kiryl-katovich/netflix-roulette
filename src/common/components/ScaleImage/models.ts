export type ScaleImageProps = {
  src: string;
  width?: string;
  ratio?: number;
};
