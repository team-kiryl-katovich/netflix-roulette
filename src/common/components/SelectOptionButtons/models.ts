export type SelectOptionButtonProps = {
  title: string;
  options: string[];
};
