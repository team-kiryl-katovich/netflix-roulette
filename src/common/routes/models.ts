export type RouteParams = {
    id: number;
}
