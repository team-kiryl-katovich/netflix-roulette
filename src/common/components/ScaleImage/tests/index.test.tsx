import * as React from 'react';
import * as enzyme from 'enzyme';

import { ScaleImageProps } from '../models';
import { ScaleImage } from '..';

describe('[COMMON] [COMPONENTS] [SCALE_IMAGE]', () => {
  test('should render component', () => {
    const fakeProps: ScaleImageProps = {
      src: 'fakeSrc',
    };

    const component = enzyme.shallow(<ScaleImage {...fakeProps} />);
    expect(component).toMatchSnapshot();
  });
});
