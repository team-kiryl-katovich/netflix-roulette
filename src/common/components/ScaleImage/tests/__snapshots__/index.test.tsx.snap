// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMMON] [COMPONENTS] [SCALE_IMAGE] should render component 1`] = `
<div
  className="scale-image"
  style={
    Object {
      "paddingTop": "100%",
      "width": "100%",
    }
  }
>
  <img
    alt=""
    className="scale-image__image"
    src="fakeSrc"
  />
</div>
`;
